### PR TITLE
Feature/product views

### PR DESCRIPTION
**Pull request by Meise90**

### CODE REVIEW

**Review by jacoor — CHANGES_REQUESTED** [review]
Please do not remove migrations. Bring those files back. 
Migrations are like version control for DB. 
Removing those migrations here breaks database for everybody else.

File: Dshop/apps/products_catalogue/migrations/0001_initial.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Generated by Django 4.2.3 on 2023-08-06 17:58
+# Generated by Django 4.2.3 on 2023-08-23 12:47
```
Comment (jacoor) [inline Dshop/apps/products_catalogue/migrations/0001_initial.py]: What happened here?
All migrations are stored also in DB. By removing and reworking those migrations everyone else will have to do an additional job to catch up. 
Please revert those migrations changes.

Comment (jacoor) [reply Dshop/apps/products_catalogue/migrations/0001_initial.py]: Django can squash migrations, if you are interested. https://docs.djangoproject.com/en/4.2/topics/migrations/#migration-squashing

File: Dshop/apps/products_catalogue/views.py [hunk 2]
```diff
@@ -1,3 +1,15 @@
-#from django.shortcuts import render
+from django.shortcuts import render
+from django.views import View
+from .models import Product
 
 # Create your views here.
+class ProductListView(View):
```
Comment (jacoor) [inline Dshop/apps/products_catalogue/views.py]: Please change to ListView
https://akademiait.com.pl/class-based-views-w-django-wygodny-sposob-tworzenia-widokow-czesc-2/

File: Dshop/apps/products_catalogue/views.py [hunk 3]
```diff
@@ -1,3 +1,15 @@
...
+    template_name = 'products_catalogue/products_list.html'
+    model = Product
+    def get(self, request):
+        all_products = Product.objects.all()
```
Comment (jacoor) [inline Dshop/apps/products_catalogue/views.py]: Products have "is_active = models.BooleanField(default=True)" field, which is ignored here. 
Please fix.

File: Dshop/templates/base.html [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        </div>
+    </div>
+</nav>
+
+{% block content %}
+
+{% endblock %}
+
+<footer class="bg-dark text-light text-center py-3">
+    <p>&copy; {{ year }} PyMasters.pl. All rights reserved.</p>
```
Comment (jacoor) [inline Dshop/templates/base.html:40]: :-)

**Review by jacoor — CHANGES_REQUESTED** [review]

File: Dshop/apps/products_catalogue/migrations/0001_initial.py (hunk 1 above)
Comment (Meise90) [reply Dshop/apps/products_catalogue/migrations/0001_initial.py]: migrations reverted :)

File: Dshop/apps/products_catalogue/views.py (hunk 2 above)
Comment (Meise90) [reply Dshop/apps/products_catalogue/views.py]: fixed :)

File: Dshop/apps/products_catalogue/views.py (hunk 3 above)
Comment (Meise90) [reply Dshop/apps/products_catalogue/views.py]: done! :)

File: pyproject.toml [hunk 5]
```diff
@@ -0,0 +1,14 @@
+[tool.poetry]
```
Comment (jacoor) [inline pyproject.toml]: We already have this file, please remove.
Ours is located in here
https://github.com/pymasterspl/Dshop/blob/dev/Dshop/pyproject.toml

File: Dshop/templates/base.html (hunk 4 above)
Comment (Meise90) [reply Dshop/templates/base.html:40]: global data passed to all templates using the context_processors.py module (in this case - 'year') :)

File: pyproject.toml (hunk 5 above)
Comment (Meise90) [reply pyproject.toml]: fixed :)

**Review by jacoor — APPROVED** [review]
Good job!

**Pull request merged**